### PR TITLE
Fix: Cannot delete favorited PS, PDS, or member

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

- Sync tree nodes in favorites list with their actual state under "Data Sets" panel

## Release Notes

Milestone: 2.5.0

Changelog: Fixed issue where Favorites section was not refreshing after changes were made to data sets or their members.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found

## Further comments

- **Future Proposal:** Change method of storing favorites such that the nodes can be mapped 1:1; this should shorten lookup time for corresponding nodes and make logic more straightforward for adding/removing favorites. Since we are searching based on labels, we could assign the node labels as keys for a map-based structure.
- **For next standup:** The `findFavoritedNode` and `findNonFavoritedNode` functions are marked deprecated, but they're being used in the codebase. Reimplement (as suggested above) or keep the same data structure?

### CODE REVIEW

**Review by JillieBeanSim — DISMISSED** [review]
This works great! thanks @traeok for the fix

File: packages/zowe-explorer/src/dataset/actions.ts [hunk 1]
```diff
@@ -438,6 +428,17 @@ export async function createMember(
             true,
             datasetProvider
         );
+
+        // Refresh corresponding tree parent to reflect addition
+        const findCorrespondingNode = contextually.isFavorite(parent)
+            ? datasetProvider.findNonFavoritedNode.bind(datasetProvider)
+            : datasetProvider.findFavoritedNode.bind(datasetProvider);
```
Comment (t1m0thyj) [inline packages/zowe-explorer/src/dataset/actions.ts:435]: Can we call the `findEquivalentNode` method here to reduce duplication?

File: packages/zowe-explorer/src/dataset/actions.ts [hunk 2]
```diff
@@ -1046,11 +1049,25 @@ export async function deleteDataset(node: api.IZoweTreeNode, datasetProvider: ap
...
+    datasetProvider.removeFavorite(node);
+
+    const isMember = contextually.isDsMember(node);
+
+    // If the node is a dataset member, go up a level in the node tree
+    // to find the relevant, matching node
+    const nodeOfInterest = isMember ? node.getParent() : node;
+    const parentNode = fav
+        ? datasetProvider.findNonFavoritedNode(nodeOfInterest)
+        : datasetProvider.findFavoritedNode(nodeOfInterest);
```
Comment (t1m0thyj) [inline packages/zowe-explorer/src/dataset/actions.ts:1064]: Can we call the `findEquivalentNode` method here to reduce duplication (same question as above)?

File: packages/zowe-explorer/src/dataset/actions.ts (hunk 1 above)
Comment (traeok) [reply packages/zowe-explorer/src/dataset/actions.ts]: Yep! Not sure how I missed that, I added the wrapper function but didn't update the logic to use it 🤦‍♂️

**Review by t1m0thyj — APPROVED** [review]
Tested these changes and the favorites tree always seems to stay in sync with other data set trees now 👍

**Review by zFernand0 — DISMISSED** [review]
TL;DR: LGTM! 😋 

At first I was afraid that `IZoweNodeType -> IZoweTreeNode` might be a breaking change to extenders, but after some trial and error, it seems that this should be fine as long as every type in `IZoweNodeType` extends `IZoweTreeNode` (which they are).

**Comment by traeok** [comment]
Sorry @zFernand0 - had to re-request review as the rebase marked previous reviews as stale..

Is there a way to avoid this in the future or is it just an inevitable result of rebasing/updating branches to reflect main?

**Comment by zFernand0** [comment]
> had to re-request review as the rebase marked previous reviews as stale. Is there a way to avoid this in the future or is it just an inevitable result of rebasing/updating branches to reflect main?

Thanks for sticking with a Rebase model 😋 
It seems like it's either dismiss reviews automatically after new commits or don't 😢 
image

**Comment by zFernand0** [comment]
If we want to keep the history clean, we could do the same thing that the `zowe/docs-site` does.
- Disallow Merge Commits (in favor of squash merging)
image

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋

**Review by t1m0thyj — APPROVED** [review]
